cmd/packet-capture.go

```diff
@@ -0,0 +1,134 @@
+package cmd
+
+import (
+	"net"
+	"os"
+	"os/exec"
+	"time"
+
+	"github.com/fatih/color"
+	"github.com/jpillora/sizestr"
+	"github.com/rodaine/table"
+	"github.com/spf13/cobra"
+)
+
+var pktCmd = &cobra.Command{
+	Use:   "get-packets",
+	Short: "",
+	Long:  "",
+	Run:   runPacketCapture,
+}
+
+type PcapResult struct {
+	NodeName    string
+	PacketCount int64
+	ByteCount   int64
+}
+
+var results = []PcapResult{}
+
+func runPacketCapture(cmd *cobra.Command, args []string) {
+	wg.Add(len(addresses))
+	for i, _ := range addresses {
+		go runPacketCaptureOnAddr(addresses[i], nodes[i])
+	}
+	wg.Wait()
+}
+
+func runPacketCaptureOnAddr(addr string, filename string) {
+	log.Infof("Starting Packet Capture for %s...", filename)
+
+	tcpServer, err := net.ResolveTCPAddr("tcp", addr)
+
+	if err != nil {
+		log.Error("ResolveTCPAddr failed:", err.Error())
+		log.Fatal(err)
+	}
+	conn, err := net.DialTCP("tcp", nil, tcpServer)
+	if err != nil {
+		log.Error("Dial failed:", err.Error())
+		log.Fatal(err)
+	}
+	var f *os.File
+	err = os.MkdirAll("./output/pcap/", 0700)
+	if err != nil {
+		log.Errorf("Create directory failed: %v", err.Error())
+		log.Fatal(err)
+	}
+	f, err = os.Create("./output/pcap/" + filename)
+	if err != nil {
+		log.Errorf("Create file %s failed: %v", filename, err.Error())
+		log.Fatal(err)
+	}
+	defer CleanupCapture(conn, f)
+	for {
+		received := make([]byte, 65535)
+		n, err := conn.Read(received)
+		if err != nil {
+			log.Error("Read data failed:", err.Error())
+			log.Fatal(err)
+		}
+		_, err = f.Write(received[:n])
+		if err != nil {
+			log.Fatal(err)
+		}
+		go managePcapTable(PcapResult{NodeName: filename, PacketCount: 1, ByteCount: int64(n)})
```

Comment:
Not sure how useful this will be by only having node, packet and byte counts, even with a filter applied.